packages/flutter_tools/lib/src/commands/build_web.dart

```diff
@@ -113,6 +114,16 @@ class BuildWebCommand extends BuildSubCommand {
         r'Please add `<base href="$FLUTTER_BASE_HREF">` to web/index.html'
       );
     }
+
+    // Currently supporting options [output-dir] and [output]
+    // as valid approaches for setting output directory of build artifacts
+    final String? outputDirectoryPath;
+    if (stringArg('output-dir') != null) {
```

Comment:
When I tried this locally, providing `flutter build web --output-dir=some-path` and then read `stringArg('output')`, I was able to see `'some-path'`

Reply:
(note I'm querying for the real name which provides the alias result--the transitive may not be true)

Reply:
hmm yeah looks like I must have been mistaken.. when i called `stringArg('output')`, it correctly retrieves the correct path regardless of whether i used the `output-dir` or `output` option

I also realized I was getting some weird behavior because of the two different dart versions on my computer as we discussed today which were cleared after removing the homebrew installed dart

Great catch!